app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java

```diff
@@ -162,9 +162,13 @@ private void checkBackAfterMissedTime(long alarmIn) {
 
     // alarmIn is relative time ms
     public void setAlarm(long alarmIn, boolean force) {
-        if (!force && (alarmIn < 5 * 60 * 1000)) {
-            // No need to check more than once every 5 minutes
-            alarmIn = 5 * 60 * 1000;
+        int aggressive_timer;
```

Comment:
why? I just replacing a fix value with another fix value, which user can define. I use these changes since several weeks and service is still working properly.

The problem with the 5 minutes is that BG values are coming every 5min and it's really likely that service will be restarted during getting next values. 

Otherwise we should use at least another value than 5.

What do you think?

Reply:
The missed readings service is used for a lot more than the aggressive restart timer and is triggered in multiple different ways and there are several callers even to the `setAlarm()` method. You can't control when it is going to fire.

If the problem is that you don't want the aggressive restart timer to restart the service at the exact moment you think a reading could come in, then you should instead target specifically the `isCollecting()` method in the collector itself.

The whole point of this method is as a flag to tell the aggressive restart not to activate if the collector is collecting. I would say the best way to do this would be to make it conditionally return that it is collecting if the time since the last reading is below some "very stale" threshold and if the modulus of the time since last data is within some window, eg +- 10 seconds of the 5 minute window for the previous collection.

Reply:
ok, will revert